### PR TITLE
Ensure all extra attributes are removed on destroy

### LINKED ISSUE #702: .destroy() question
Hi,

the .destroy() call should remove all extra stuff the editor bind to the editable DOM element?
Because for me the
role="textbox"
aria-multiline="true" 
medium-editor-index="0"

Stays inthe code (I tried on the demo page via console and there I had the same result).

Is it a feature ( because of course I can remove them manually) or some minor bug?

Thx.

John

### PR DESCRIPTION
**Pull request by j0k3r**
Fix #702

### CODE REVIEW

File: src/js/core.js [hunk 1]
```diff
@@ -520,9 +520,14 @@ function MediumEditor(elements, options) {
                     element.innerHTML = element.innerHTML;
                 }
 
+                // cleanup extra added attributes
                 element.removeAttribute('contentEditable');
                 element.removeAttribute('spellcheck');
+                element.removeAttribute('data-medium-focused');
                 element.removeAttribute('data-medium-editor-element');
+                element.removeAttribute('role');
```
Comment (nmielnik) [inline src/js/core.js]: I would vote for only removing attributes that we knowingly added. I'm not sure why `role` and `aria-multiline` are getting added by some browsers, but if someone purposefully added those attributes before initializing and destroying MediumEditor, I don't think we should remove them.

So yeah, can we reduce this to only remove `data-medium-focused` and `medium-editor-index`?

File: spec/setup.spec.js [hunk 2]
```diff
@@ -31,6 +31,10 @@ describe('Setup/Destroy TestCase', function () {
             spyOn(MediumEditor.prototype, 'setup').and.callThrough();
             editor.setup();
             expect(editor.setup).toHaveBeenCalled();
+            expect(document.querySelector('[data-medium-editor-element]')).toBeTruthy();
+            expect(document.querySelector('[aria-multiline]')).toBeTruthy();
+            expect(document.querySelector('[medium-editor-index]')).toBeTruthy();
+            expect(document.querySelector('[role]')).toBeTruthy();
```
Comment (nmielnik) [inline spec/setup.spec.js]: Could we just check for all of the attributes here (including `spellcheck` and `contentEditable`)?

File: spec/setup.spec.js [hunk 3]
```diff
@@ -44,6 +48,12 @@ describe('Setup/Destroy TestCase', function () {
             expect(document.querySelector('.medium-editor-toolbar')).toBeTruthy();
             editor.destroy();
             expect(document.querySelector('.medium-editor-toolbar')).toBeFalsy();
+
+            expect(document.querySelector('[data-medium-editor-element]')).toBeFalsy();
+            expect(document.querySelector('[data-medium-focused]')).toBeFalsy();
+            expect(document.querySelector('[aria-multiline]')).toBeFalsy();
+            expect(document.querySelector('[medium-editor-index]')).toBeFalsy();
+            expect(document.querySelector('[role]')).toBeFalsy();
```
Comment (nmielnik) [inline spec/setup.spec.js]: If we did this check here, we could ensure we don't add any other stray attributes in the future:

```js
expect(this.el.getAttribute('class')).toBe('editor');
expect(this.el.attributes.length).toBe(1); // 'class' should be the only remaining attribute
```

File: spec/setup.spec.js (hunk 2 above)
Comment (j0k3r) [inline spec/setup.spec.js]: Done

**Pull request merged**